regparser/diff/tree.py

```diff
@@ -0,0 +1,99 @@
+from regparser.diff.text import get_opcodes
+from regparser.tree import struct
+
+
+ADDED = 'added'
+MODIFIED = 'modified'
+DELETED = 'deleted'
+
+
+def _local_changes(lhs, rhs):
+    """Account for only text changes between nodes. This explicitly excludes
+    children"""
+    if lhs.text == rhs.text and lhs.title == rhs.title:
+        return []
+    else:
+        node_changes = {"op": MODIFIED}
+
+        text_opcodes = get_opcodes(lhs.text, rhs.text)
+        if text_opcodes:
+            node_changes["text"] = text_opcodes
+
+        title_opcodes = get_opcodes(lhs.title, rhs.title)
+        if title_opcodes:
+            node_changes["title"] = title_opcodes
+        return [(lhs.label_id, node_changes)]
+
+
+def _new_in_rhs(lhs_list, rhs_list):
+    """Compare the lhs and rhs lists to see if the rhs contains elements not
+    in the lhs"""
+    added = []
+    lhs_codes = tuple(map(lambda n: n.label_id, lhs_list))
+    for node in rhs_list:
+        if node.label_id not in lhs_codes:
+            added.append(node)
+    return added
+
+
+def _index_by_label_id(nodes):
+    # Replace with dict comprehension when not on 2.6
+    return dict((n.label_id, n) for n in nodes)
+
+
+def _data_for_add(node):
+    node_as_dict = {
+        'child_labels': tuple(c.label_id for c in node.children),
+        'label': node.label,
+        'node_type': node.node_type,
+        'tagged_text': node.tagged_text or None,  # maintain backwards compat
+        'text': node.text,
+        'title': node.title or None,
+    }
+    return (node.label_id, {"op": ADDED, "node": node_as_dict})
+
+
+def _data_for_delete(node):
+    return (node.label_id, {"op": DELETED})
+
+
+def changes_between(lhs, rhs):
+    """Main entry point for this library. Recursively return a list of changes
+    between the lhs and rhs. lhs and rhs should be FrozenNodes. Note that this
+    *does not* account for reordering nodes, though it does account for
```

Comment:
We currently don't implement reordering in diffs (with a few exceptions). I don't know if that's acceptable or not, just noted it.